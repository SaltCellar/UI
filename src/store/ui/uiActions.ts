import { createAction } from 'typesafe-actions';

export const toggleSidebar = createAction('ui/toggle_sidebar');
