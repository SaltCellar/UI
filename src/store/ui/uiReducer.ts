import { ActionType, getType } from 'typesafe-actions';
import { toggleSidebar } from './uiActions';

export type UIAction = ActionType<typeof toggleSidebar>;

export type UIState = Readonly<{
  isSidebarOpen: boolean;
}>;

export const defaultState: UIState = {
  isSidebarOpen: false,
};

export const stateKey = 'ui';

export function uiReducer(state = defaultState, action: UIAction): UIState {
  switch (action.type) {
    case getType(toggleSidebar):
      return {
        ...state,
        isSidebarOpen: !state.isSidebarOpen,
      };
    default:
      return state;
  }
}
