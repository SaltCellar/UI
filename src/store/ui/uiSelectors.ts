import { RootState } from '../rootReducer';
import { stateKey } from './uiReducer';

export const selectUIState = (state: RootState) => state[stateKey];

export const selectIsSidebarOpen = (state: RootState) =>
  selectUIState(state).isSidebarOpen;
