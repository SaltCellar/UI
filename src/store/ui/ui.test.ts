jest.mock('api/users');

import { uiSelectors } from '.';
import { createMockStoreCreator } from '../mockStore';
import * as actions from './uiActions';
import { stateKey, uiReducer } from './uiReducer';
import * as selectors from './uiSelectors';

const createUIStore = createMockStoreCreator({
  [stateKey]: uiReducer,
});

test('default state', async () => {
  const store = createUIStore();
  expect(selectors.selectUIState(store.getState())).toMatchSnapshot();
});

test('toggle sidebar', async () => {
  const store = createUIStore();
  store.dispatch(actions.toggleSidebar());
  expect(uiSelectors.selectIsSidebarOpen(store.getState())).toBe(true);
});
