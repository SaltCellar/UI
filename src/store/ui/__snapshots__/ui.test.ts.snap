// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default state 1`] = `
Object {
  "isSidebarOpen": false,
}
`;
